File: src/mffs/event/BlockDropDelayedEvent.java
package mffs.event;

import mffs.DelayedEvent;
import mffs.IDelayedEventHandler;
import net.minecraft.block.Block;
import net.minecraft.world.World;
import universalelectricity.core.vector.Vector3;

public class BlockDropDelayedEvent extends DelayedEvent
{
	protected Block block;
	protected World world;
	protected Vector3 position;

	public BlockDropDelayedEvent(IDelayedEventHandler handler, int ticks, Block block, World world, Vector3 position)
	{
		super(handler, ticks);
		this.block = block;
		this.world = world;
		this.position = position;
	}

	@Override
	protected void onEvent()
	{
		if (!this.world.isRemote)
		{
			if (this.position.getBlockID(this.world) == this.block.blockID)
			{
				this.block.dropBlockAsItem(this.world, this.position.intX(), this.position.intY(), this.position.intZ(), this.position.getBlockMetadata(world), 0);
				this.position.setBlock(this.world, 0);
			}
		}
	}
}

File: src/main/scala/mffs/field/mobilize/event/BlockDropDelayedEvent.scala
package mffs.field.mobilize.event

import nova.core.block.Block
import nova.core.game.Game
import nova.core.util.transform.Vector3i
import nova.core.world.World

import scala.collection.convert.wrapAll._

class BlockDropDelayedEvent(ticks: Int, block: Block, world: World, position: Vector3i) extends DelayedEvent(ticks) {
	protected override def onEvent {
		if (Game.instance.networkManager.isServer) {
			val checkBlock = world.getBlock(position)

			if (checkBlock.isPresent && checkBlock.get == block) {
				val drops = block.getDrops
				drops.foreach(drop => world.createEntity(position.toDouble + 0.5, drop))
				world.setBlock(position, Game.instance.blockManager.getAirBlock)
			}
		}
	}
}